spirv_msl.cpp

```diff
@@ -5002,6 +5002,33 @@ void CompilerMSL::emit_custom_functions()
 			statement("");
 			break;
 
+		case SPVFuncImplQuantizeToF16:
+			// Ensure fast-math is disabled to match Vulkan results.
+			statement("[[clang::optnone]] float spvQuantizeToF16(float val)");
```

Comment:
Can't this be made into a template?

Reply:
I've made this into a template now. It's still a bit messy because of the need for a separate struct template to align the `half*` with the `float*` types, but it was the most suitable option I could come up with for accomplishing that. I'm happy to hear alternate approaches to handling that mapping.

I had originally tried to come up with a template, but gave up because of that. Thanks for the prod to force me to have a second look at the problem.

Reply:
Partial specialization with an NTTP:
```c++
template<typename F> struct spvHalfTypeSelector;
template<uint N> struct spvHalfTypeSelector<vec<float, N>> { using H = vec<half, N>; };
```
This should make it a little less messy.

Reply:
Thanks for thinking of `vec<T, n>`. It's a clever solution for n > 1, and saves a couple of template lines.